docs/source/user-guide/sql/subqueries.md

```diff
@@ -19,80 +19,392 @@
+in the results of a _[correlated subquery](#correlated-subqueries)_.
+`NOT IN` returns all rows where a given expression’s value cannot be found in
+the results of a subquery or list of values.
+
+```sql
+expression [NOT] IN (subquery|list-literal)
+```
+
+#### Examples
+
+```sql
+SELECT * FROM x WHERE column_1 IN (1,3);
+
 +----------+----------+
 | column_1 | column_2 |
 +----------+----------+
 | 1        | 2        |
 +----------+----------+
-1 row in set.
 ```
 
-## NOT EXISTS
+```sql
+SELECT * FROM x WHERE column_1 NOT IN (1,3);
+
++----------+----------+
+| column_1 | column_2 |
++----------+----------+
+| 2        | 4        |
++----------+----------+
+```
+
+## SELECT clause subqueries
+
+`SELECT` clause subqueries use values returned from the inner query as part
+of the outer query's `SELECT` list.
+The `SELECT` clause only supports [scalar subqueries](#scalar-subqueries) that
+return a single value per execution of the inner query.
+The returned value can be unique per row.
+
+```sql
+SELECT [expression1[, expression2, ..., expressionN],] (<subquery>)
+```
+
+**Note**: `SELECT` clause subqueries can be used as an alternative to `JOIN`
+operations.
 
-The `NOT EXISTS` syntax can be used to find all rows in a relation where a correlated subquery produces zero matches
-for that row. Only correlated subqueries are supported.
+### Example
 
 ```sql
-select * from x y where not exists (select * from x where x.column_1 = y.column_1);
-0 rows in set.
+SELECT
+  column_1,
+  (
+    SELECT
+      first_value(string)
+    FROM
+      y
+    WHERE
+      number = x.column_1
+  ) AS "numeric string"
+FROM
+  x;
+
++----------+----------------+
+| column_1 | numeric string |
++----------+----------------+
+|        1 | one            |
+|        2 | two            |
++----------+----------------+
 ```
 
-## IN
+## FROM clause subqueries
 
-The `IN` syntax can be used to find all rows in a relation where a given expression's value can be found in the
-results of a correlated subquery.
+`FROM` clause subqueries return a set of results that is then queried and
+operated on by the outer query.
 
 ```sql
-select * from x where column_1 in (select column_1 from x);
+SELECT expression1[, expression2, ..., expressionN] FROM (<subquery>)
+```
+
+### Example
+
+The following query returns the average of maximum values per room.
+The inner query returns the maximum value for each field from each room.
+The outer query uses the results of the inner query and returns the average
+maximum value for each field.
+
+```sql
+SELECT
+  column_2
+FROM
+  (
+    SELECT
+      *
+    FROM
+      x
+    WHERE
+      column_1 > 1
+  );
+
++----------+
+| column_2 |
++----------+
+|        4 |
++----------+
+```
+
+## WHERE clause subqueries
+
+`WHERE` clause subqueries compare an expression to the result of the subquery
+and return _true_ or _false_.
+Rows that evaluate to _false_ are filtered from results.
```

Comment:
Technically it is rows that do not evaluate to `_true` (aka rows that evaluate to `NULL` are also filtered)

```suggestion
Rows that evaluate to _false_ or `_NULL_` are filtered from results.
```